ggml.c

```diff
@@ -10158,12 +10201,28 @@ static void ggml_compute_forward_mul_mat(
     const int64_t blck_0 = 16;
     const int64_t blck_1 = 16;
 
+    // dot kernels can handle 1 row and col at a time, but mmla kernels can process 2 rows and cols
+    int64_t nrc = vec_dot_num_rows;
+    // TODO: currently the mmla kernels support only even numbered rows/cols.
+    // this check can be removed once they are extended to support odd numbered rows/cols too
+    if ((nr0 % 2 != 0) || (nr1 % 2 != 0)) {
+        nrc = 1;
+    }
+    const int64_t i13s = (1/(ne12*ne1));
+    const int64_t i12s = (1 - i13s*ne12*ne1)/ne1;
+    const int64_t i11s = (1 - i13s*ne12*ne1 - i12s*ne1);
+
+    const int64_t src1_col_stride = (src1_cont || src1->type != vec_dot_type
+                     ? (i11s + i12s*ne11 + i13s*ne12*ne11 )*row_size
+                     : (i11s*nb11 + i12s*nb12  + i13s*nb13));
+
```

Comment:
The stride calculation can be simplified here.

But I also think this needs stronger assert, because `nr1 % 2 == 0` is not enough.
I think it needs to be `ne11 % 2 == 0`, because even if `src1` has even number of total rows, it could have odd number of rows across dim 1 and for non-contiguous 3D tensors, the stride between rows `[...][ne11-1][0]` and `[...][0][1]` could be different

Reply:
thanks, I'll use ` ne11` for the even rows constraint. 
regarding  the stride calculation,  If I understand correctly, this can't be derived from the `nb[]` like I did for src0 and dst, that's why I had to use the logic similar to offset calculation. am I missing something? do you have any pointers?

Reply:
thanks, @ggerganov , the simplified stride calculation worked, last time I tried `nb11` for both contiguous and non-contiguous buffers, that's why it didn't work. I have also fixed the stride passing logic to vec_dot() call, so that kernels with any number of rows can reuse this.

Reply:
I have updated the PR